cmd/modelplex/main_server_test.go

```diff
@@ -0,0 +1,364 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/modelplex/modelplex/internal/config"
+	"github.com/modelplex/modelplex/internal/server"
+)
+
+func TestHTTPServerByDefault(t *testing.T) {
+	// Create a test config
+	cfg := &config.Config{
+		Providers: []config.Provider{
+			{
+				Name:     "test",
+				Type:     "openai",
+				BaseURL:  "http://localhost:8080",
+				APIKey:   "test-key",
+				Models:   []string{"test-model"},
+				Priority: 1,
+			},
+		},
+		Server: config.Server{
+			LogLevel:       "info",
+			MaxRequestSize: 1024,
+		},
+	}
+
+	// Test HTTP server creation
+	srv := server.NewWithHTTP(cfg, "127.0.0.1", 0) // Use port 0 to get a random available port
+	
+	// Start server in goroutine
+	go func() {
+		if err := srv.Start(); err != nil && err != http.ErrServerClosed {
+			t.Errorf("Failed to start HTTP server: %v", err)
+		}
+	}()
```

Comment:
Prefer `errgroup.Go`, for automatic synchronization with `errgroup.Wait`. Here and in every other test.

```go
eg, _ := errgroup.WithContext(t.Context())
eg.Go(srv.Start())
err := eg.Wait()
```